sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxGatewayStoreModel.java

```diff
@@ -344,7 +344,17 @@ private String ensureSlashPrefixed(String path) {
                        if (!(exception instanceof CosmosException)) {
                            // wrap in CosmosException
                            logger.error("Network failure", exception);
-                           dce = BridgeInternal.createCosmosException(request.requestContext.resourcePhysicalAddress, 0, exception);
+
+                           int statusCode = 0;
+                           if (WebExceptionUtility.isNetworkFailure(exception)) {
+                               if (WebExceptionUtility.isReadTimeoutException(exception)) {
```

Comment:
We have these if checks below , can we please use those to update the status code ?

Reply:
No - because the ones below are applicaple whether the original exception is ComsosException or not - while setting the status code is only relevant when the original exception is not a CosmosException.